subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -2290,21 +2301,25 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
 						  0, 0, ticker_update_op_cb);
 
 #if defined(CONFIG_BT_CTLR_ADV_EXT)
-		adv->event_counter += (lazy + 1);
-
-		if (adv->ticks_remain_duration) {
-			uint32_t ticks_interval =
-				HAL_TICKER_US_TO_TICKS((uint64_t)adv->interval *
-						       ADV_INT_UNIT_US);
-			uint32_t ticks_elapsed = ticks_interval * (lazy + 1) +
-						 ticks_drift;
-
-			if (adv->ticks_remain_duration > ticks_elapsed) {
-				adv->ticks_remain_duration -= ticks_elapsed;
+		if (adv->remain_duration_us && adv->event_counter > 0) {
```

Comment:
Why `adv->event_counter > 0` ?

If say, first 2 events where skipped due to overlap, then `adv->remain_duration_us` would not be reduced as we will not enter this if-clause.

```suggestion
		adv->event_counter += (lazy + 1);
		
		if (adv->remain_duration_us) {
```

Reply:
According to the spec (vol 4, part E, section 7.8.56):
> The Duration[i] parameter indicates the duration for which that advertising set
is enabled. The duration begins at the start of the first advertising event of this
advertising set.

Before, we would actually end the advertising one event too early since we subtracted one interval when the first advertising event was started.

The skipping of events is a bit problematic, yes; I'm not sure what the correct thing to do is there. On one hand you could argue that the first advertising event is the one actually sent out on the radio; On the other hand that would lead to less predictable behavior for the host, so maybe it should be counted as if they were sent.